rama-tcp/src/client/pool.rs

```diff
@@ -0,0 +1,150 @@
+use std::sync::{atomic::AtomicUsize, Arc};
+
+use rand::seq::SliceRandom;
+
+use super::TcpStreamConnector;
+
+/// A pool of TcpConnectors
+#[derive(Debug, Clone)]
+pub struct TcpStreamConnectorPool<C> {
+    idx: Arc<AtomicUsize>,
+    connectors: Vec<C>,
+}
+
+impl<C: TcpStreamConnector> TcpStreamConnectorPool<C> {
+    /// A `TcpStreamConnector` where each connection is chosed randomly from a pool of
+    /// `TcpStreamConnector`s
+    pub fn new_random(connectors: Vec<C>) -> Self {
+        Self::new_random_with_rng(connectors, rand::rng())
+    }
+
+    fn shuffle_connectors<Rng: rand::Rng>(mut connectors: Vec<C>, mut rng: Rng) -> Vec<C> {
```

Comment:
why is this a method? or even a thing at all, you are basically just calling shuffle

Reply:
Fair point. There was an earlier version of the code where the unit test would call this function. I'll remove it.